### DISCUSSION TITLE
ESP32 - static IP address for ethernet port

### DISCUSSION THREAD

**Discussion by thermseekr** [arendst/Tasmota#11166] [Q&A] [answered]
Hello all,

Is there a way to set a static IP address for the ethernet port on the ESP32? I've been playing with an Olimex ESP32-POE-ISO and it works just great, I only miss being able to set the static address.

I haven't found it in any of the configuration files, there's something commented out in xdrv_82 but I'm not sure what it's supposed to mean so I left it alone.

Thank you very much for spending time on the ESP32 version, having tasmota on a cabled connection was my fondest home automation dream.

Regards,
Tales

**Comment by barbudor** [comment, marked as answer]
Hi
I don't see any code commented related to IP address
However the framework function to set ip addresses seems to be
```
        bool config(IPAddress local_ip, IPAddress gateway, IPAddress subnet, IPAddress dns1 = (uint32_t)0x00000000, IPAddress dns2 = (uint32_t)0x00000000);
```
from `lib\libesp32\ESP32-Ethernet\src\ETH.h`

If would expect the proper location to use it is in `EthernetInit()`, once after `ETH.begin()` call.
Something like:
```
if ( Settings.ipv4_address[0] ) {
  if (!ETH.config( Settings.ipv4_address[0], Settings.ipv4_address[1], Settings.ipv4_address[2], Settings.ipv4_address[3] )) {
    AddLog(LOG_LEVEL_DEBUG, PSTR("ETH: Error with setting static IP"));
  };
}
```

That's a guess but very similar to `support_wifi.inio:206`
I don't have any ESP32-ETH to test but if you succeed I would expect a PR to be welcome.

**Reply by thermseekr** [reply]
Hi,

Thanks for you reply. I don't feel qualified enough to submit PRs to this project, but will take a look at this tonight.

Cheers

**Reply by barbudor** [reply]
I just wonder if my suggestion is fully ok.
As `IpAddress[0]` is the Wifi IP address, I am not sure that Wifi is disabled when using ETH.
If Wifi is disabled, then it should be ok to use `IpAddress[0]`
Otherwise there will be an internal conflict with 2 interfaces on the same IP address....
Let me know your findings

**Reply by barbudor** [reply]
Just get confirmed that it won't work as Wifi is not de-activated and will use Ipaddress1 (IpAddress[0])
That will be an internal IP address conflict

**Reply by barbudor** [reply]
Is that mandatory to use Static ? Couldn't you configure your DHCP to assign a given address based on ETH MAC address ?

**Reply by sfromis** [reply]
While the potential address conflict should be avoidable using the ESP32 `Wi-Fi` command to disable Wi-Fi, I'm not really expecting IpAddress1 to be made available to wired networking. You could try if it works....

**Reply by thermseekr** [reply]
> Is that mandatory to use Static ? Couldn't you configure your DHCP to assign a given address based on ETH MAC address ?

No it's not mandatory, I can setup a DHCP reservation, I actually did it. Only I was wondering if there was some config file as it's the case for WiFi in my_user_config.h, I thought maybe I'm not looking in the right place.

**Comment by thermseekr** [comment]
In case someone comes to this topic, this subject has been solved in #14385 
Thanks a lot @barbudor and @arendst
